app.py

```diff
@@ -119,13 +133,33 @@ def post(self):
         self.finish(json.dumps(finish_ret))
 
 
+def get_user_img_prefix(self):
+    cookie = self.get_cookie(app_cookie)
+    user_id = cookie if cookie else ""
+    return temp_img_prefix + user_id + "-"
+
+
+def valid_user_img(self, img):
+    default_img = not img.startswith(static_img_path + temp_img_prefix)
+    user_img = img.startswith(static_img_path + get_user_img_prefix(self))
+    current_user_img = user_img if self.get_cookie(app_cookie) else False
+    return default_img or current_user_img
+
+
+def get_image_captions(self):
+    return collections.OrderedDict(
+        (k, v) for k, v in image_captions.items() if valid_user_img(self, k)
+    )
+
+
 def run_ml_queued(img_path, ret_queue):
     caption = run_ml(img_path)
     ret_queue.put((img_path, caption))
 
 
 def valid_file_ext(filename):
-    return '.' in filename and filename.split('.', 1)[1].lower() in VALID_EXT
+    filename, file_extension = os.path.splitext(filename)
```

Comment:
nit, unused variables part of tuples should be prefixed with an underscore. `_filename, file_extension`

Reply:
didn't know that, I'd seen that before but didn't know why